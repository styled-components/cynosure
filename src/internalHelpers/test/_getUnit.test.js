// @flow

import getUnit from '../_getUnit'

describe('getUnit', () => {
  it('should get px from whole values', () => {
    expect(getUnit('1px')).toMatchSnapshot()
  })

  it('should get px from values', () => {
    expect(getUnit('1.5px')).toMatchSnapshot()
  })

  it('should get pt from whole values', () => {
    expect(getUnit('1pt')).toMatchSnapshot()
  })

  it('should get pt from values', () => {
    expect(getUnit('1.5pt')).toMatchSnapshot()
  })

  it('should get pc from whole values', () => {
    expect(getUnit('1pc')).toMatchSnapshot()
  })

  it('should get pc from values', () => {
    expect(getUnit('1.5pc')).toMatchSnapshot()
  })

  it('should get mm from whole values', () => {
    expect(getUnit('1mm')).toMatchSnapshot()
  })

  it('should get mm from values', () => {
    expect(getUnit('1.5mm')).toMatchSnapshot()
  })

  it('should get q from whole values', () => {
    expect(getUnit('1q')).toMatchSnapshot()
  })

  it('should get q from values', () => {
    expect(getUnit('1.5q')).toMatchSnapshot()
  })

  it('should get cm from whole values', () => {
    expect(getUnit('1cm')).toMatchSnapshot()
  })

  it('should get cm from values', () => {
    expect(getUnit('1.5cm')).toMatchSnapshot()
  })

  it('should get in from whole values', () => {
    expect(getUnit('1in')).toMatchSnapshot()
  })

  it('should get in from values', () => {
    expect(getUnit('1.5in')).toMatchSnapshot()
  })

  it('should get em from whole value', () => {
    expect(getUnit('1em')).toMatchSnapshot()
  })

  it('should get em from decimal values', () => {
    expect(getUnit('1.2em')).toMatchSnapshot()
  })

  it('should get rem from whole values', () => {
    expect(getUnit('1rem')).toMatchSnapshot()
  })

  it('should get rem from decimal values', () => {
    expect(getUnit('1.2rem')).toMatchSnapshot()
  })

  it('should get ex from whole values', () => {
    expect(getUnit('1ex')).toMatchSnapshot()
  })

  it('should get ex from decimal values', () => {
    expect(getUnit('1.2ex')).toMatchSnapshot()
  })

  it('should get ch from whole values', () => {
    expect(getUnit('1ch')).toMatchSnapshot()
  })

  it('should get ch from decimal values', () => {
    expect(getUnit('1.2ch')).toMatchSnapshot()
  })

  it('should get vh from whole values', () => {
    expect(getUnit('100vh')).toMatchSnapshot()
  })

  it('should get vh from decimal values', () => {
    expect(getUnit('33.33vh')).toMatchSnapshot()
  })

  it('should get vw from whole values', () => {
    expect(getUnit('100vw')).toMatchSnapshot()
  })

  it('should get vw from decimal values', () => {
    expect(getUnit('33.33vw')).toMatchSnapshot()
  })

  it('should get vmin from whole values', () => {
    expect(getUnit('100vmin')).toMatchSnapshot()
  })

  it('should get vmin from decimal values', () => {
    expect(getUnit('33.33vmin')).toMatchSnapshot()
  })

  it('should get vmax from whole values', () => {
    expect(getUnit('100vmax')).toMatchSnapshot()
  })

  it('should get vmax from decimal values', () => {
    expect(getUnit('33.33vmax')).toMatchSnapshot()
  })

  it('should get % from whole values', () => {
    expect(getUnit('80%')).toMatchSnapshot()
  })

  it('should get % from decimal values', () => {
    expect(getUnit('33.3%')).toMatchSnapshot()
  })

  it('should return null when passed a unitless value', () => {
    expect(getUnit('100')).toMatchSnapshot()
  })

  it('should return null when passed an untouched string', () => {
    expect(getUnit('red')).toMatchSnapshot()
  })
})
