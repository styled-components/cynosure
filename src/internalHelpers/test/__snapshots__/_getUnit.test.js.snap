exports[`getUnit should get % from decimal values 1`] = `"%"`;

exports[`getUnit should get % from decimal values 2`] = `"%"`;

exports[`getUnit should get % from whole values 1`] = `"%"`;

exports[`getUnit should get ch from decimal values 1`] = `"ch"`;

exports[`getUnit should get ch from decimal values 2`] = `"ch"`;

exports[`getUnit should get ch from whole values 1`] = `"ch"`;

exports[`getUnit should get cm from values 1`] = `"cm"`;

exports[`getUnit should get cm from values 2`] = `"cm"`;

exports[`getUnit should get cm from whole values 1`] = `"cm"`;

exports[`getUnit should get em from decimal values 1`] = `"em"`;

exports[`getUnit should get em from decimal values 2`] = `"em"`;

exports[`getUnit should get em from whole value 1`] = `"em"`;

exports[`getUnit should get ex from decimal values 1`] = `"ex"`;

exports[`getUnit should get ex from decimal values 2`] = `"ex"`;

exports[`getUnit should get ex from whole values 1`] = `"ex"`;

exports[`getUnit should get in from values 1`] = `"in"`;

exports[`getUnit should get in from values 2`] = `"in"`;

exports[`getUnit should get in from whole values 1`] = `"in"`;

exports[`getUnit should get mm from values 1`] = `"mm"`;

exports[`getUnit should get mm from values 2`] = `"mm"`;

exports[`getUnit should get mm from whole values 1`] = `"mm"`;

exports[`getUnit should get pc from values 1`] = `"pc"`;

exports[`getUnit should get pc from values 2`] = `"pc"`;

exports[`getUnit should get pc from whole values 1`] = `"pc"`;

exports[`getUnit should get pt from values 1`] = `"pt"`;

exports[`getUnit should get pt from values 2`] = `"pt"`;

exports[`getUnit should get pt from whole values 1`] = `"pt"`;

exports[`getUnit should get px from values 1`] = `"px"`;

exports[`getUnit should get px from values 2`] = `"px"`;

exports[`getUnit should get px from whole values 1`] = `"px"`;

exports[`getUnit should get q from values 1`] = `"q"`;

exports[`getUnit should get q from values 2`] = `"q"`;

exports[`getUnit should get q from whole values 1`] = `"q"`;

exports[`getUnit should get rem from decimal values 1`] = `"rem"`;

exports[`getUnit should get rem from decimal values 2`] = `"rem"`;

exports[`getUnit should get rem from whole values 1`] = `"rem"`;

exports[`getUnit should get vh from decimal values 1`] = `"vh"`;

exports[`getUnit should get vh from decimal values 2`] = `"vh"`;

exports[`getUnit should get vh from whole values 1`] = `"vh"`;

exports[`getUnit should get vmax from decimal values 1`] = `"vmax"`;

exports[`getUnit should get vmax from decimal values 2`] = `"vmax"`;

exports[`getUnit should get vmax from whole values 1`] = `"vmax"`;

exports[`getUnit should get vmin from decimal values 1`] = `"vmin"`;

exports[`getUnit should get vmin from decimal values 2`] = `"vmin"`;

exports[`getUnit should get vmin from whole values 1`] = `"vmin"`;

exports[`getUnit should get vw from decimal values 1`] = `"vw"`;

exports[`getUnit should get vw from decimal values 2`] = `"vw"`;

exports[`getUnit should get vw from whole values 1`] = `"vw"`;

exports[`getUnit should return null when passed a unitless value 1`] = `null`;

exports[`getUnit should return null when passed an untouched string 1`] = `null`;
